api/src/telemetry/utils/get-role-counts-by-users.ts

```diff
@@ -0,0 +1,36 @@
+import { useEnv } from '@directus/env';
+import { LimitExceededError } from '@directus/errors';
+import type { Knex } from 'knex';
+import { getUserCount, type UserCount } from './get-user-count.js';
+
+const env = useEnv();
+
+/**
+ * Ensure that user limits are not reached
+ */
+export async function checkIncreasedUserLimits(db: Knex, increasedUserCounts: UserCount): Promise<void> {
+	if (!increasedUserCounts.admin && !increasedUserCounts.app && !increasedUserCounts.api) return;
+
+	const userCounts = await getUserCount(db);
```

Comment:
Im wondering if this is something we should cache for large database? 🤔 will do some testing on a larger DB

Reply:
Ah... Interesting! Are you thinking of caching the limits in Redis? 🤔

Reply:
Tested with 10k app users and 1k roles and this was not an immediate performance issue yet, but it does slow down and will be a good one to keep in mind as future optimization.